modal/cli/run.py

```diff
@@ -58,20 +51,39 @@ def _get_signature(f: Callable, is_method: bool = False) -> Dict[str, inspect.Pa
     return {param.name: param for param in inspect.signature(f).parameters.values()}
 
 
+def _get_param_type_as_str(annot: Any) -> str:
+    """Return annotation as a string, handling various spellings for optional types."""
+    if annot is inspect.Signature.empty:
+        return "Any"
+    annot_str = str(annot)
+    annot_patterns = [
+        r"typing.Optional\[([\w.]+)\]",
```

Comment:
Is there ever a case where this is just `Optional`? (no `typing.`?)

Reply:
Hm good point, yeah I think an annotation with postponed evaluation where `Optional` was imported directly will end up without the `typing` prefix (whereas an evaluated annotation will have the full name regardless of how it was spelled in the code). Simple enough to make it optional (heh) in the regex.

Reply:
Handled this better by evaluating the type hints